ui/src/modals/RuleModal.vue

```diff
@@ -0,0 +1,384 @@
+                app: null,
+                config: {},
+                branch: "",
+
+                // input tab
+                subject_match: "", 
+                session_match: "", 
+                extra_datatype_tags: {},
+                input_selection: {},
+                input_multicount: {},
+                input_project_override: {},
+                input_subject: {},
+                input_session: {},
+                input_tags: {},
+
+                // output tab
+                output_tags: {}, 
+                archive: {},
+            },
+
+            cb: null, // cb function to call with (err, rule[updated])
+            ws: null, // websocket
+        }
+    },
+    computed: {
+        title() {
+            if(!this.rule) return;
+            if(this.rule.name) return this.rule.name;
+            if(this.rule.app) return this.rule.app.name;
+            return "New Rule";
+        },
+
+        numJobsRunning() {
+            if(!this.rule.stats) return 0;
+            if(!this.rule.stats.tasks) return 0;
+            let count = 0;
+            for(let state in this.rule.stats.tasks) {
+                count += this.rule.stats.tasks[state];
+            }
+            return count;
+        },
+    },
+
+    mounted() {
+        this.$root.$on("rule.edit", opt=>{
+            this.tab = 0;
+            this.cb = opt.cb;
+            
+            //overwrite with user provided default
+            this.rule = Object.assign({}, 
+                this.rule,
+                getEmptyIOConfigObj(),
+                this.initializeAppIdsInRuleObj(this.rule.app, this.rule),
+                {
+                    name: "",
+                    app: null,
+                    branch: this.rule.app ? this.rule.app.github_branch : null,
+                    stats: {},
+                    active: true,
+                    _id: undefined,
+                }, 
+                { config: this.initializeAppConfigIdsInRuleConfigObj(this.rule.app ? this.rule.app.config : undefined, this.rule.config) },
+                opt.rule, // don't let anything overwrite data passed in via opt.rule
+            );
+
+            this.subscribe();
+            this.$root.$emit('bv::show::modal', 'modal-rule')
+        });
+
+        // this trigger does not actually cause the ruleModal to open. It is simply
+        // here to create rules for a given pipeline. We do that here so that we have 
+        // access to all the methods for creating a rule
+        this.$root.$on('create_pipeline', async ({ projectId, pipelineName }) => {
+            // check if we support the given pipeline first
+            const upperCasePipelineName = (pipelineName || '').toLocaleUpperCase();           
+            const pipeline = SUPPORTED_PIPELINES[upperCasePipelineName];
+            if (!pipeline || !projectId) return;
+
+            const retrievedApps = await Promise.all(pipeline.map(appId => this.$http.get(`app/${appId}`)));
```

Comment:
I believe you can do one API call and get all the apps info, you just need to adjust the fields you want to select, separated by +
```
https://brainlife.io/api/warehouse/app?select=github_branch&limit=500&find={%22_id%22:[%2258c55467e13a50849b2587ff%22]}
```

Reply:
done